### ISSUE TITLE
Error handling in CustomDict

### ISSUE THREAD

**Issue by arash-d** [alm0ra/mockafka-py#69]
The `CustomDict` class overrides the `__getitem__` method to suppress `KeyError` exceptions and return `None` if a key is not found. While this may be intentional, it could potentially mask errors in the code. It's essential to handle errors appropriately and provide meaningful feedback to users.

```python
class CustomDict(dict):
    def __getitem__(self, key):
        try:
            return super().__getitem__(key)
        except KeyError:
            return
```

https://github.com/alm0ra/mockafka-py/blob/2eb4b515775882133e029a7c959f3c1c94103037/mockafka/cluster_metadata.py#L47

**Comment by alm0ra** [comment]
Hey @arash-d, 

Thanks for highlighting this issue. Do you have any suggestions on how to address this error? Also, I'm unsure about how Kafka handles this scenario. Your contribution would be greatly appreciated.

**Comment by arash-d** [comment]
Hey @alm0ra!
Sorry for the late reply!

Actually you have 3 options:
1. Avoid try/except and using a default value `return super().__getitem__(key, None)`
2. In the case of exception raise a CustomException
3. Add logging to your except block and  `return None`, something like `logging.warning(f"Key '{key}' not found in CustomDict")`

Based on your explanation about Kafka handling in this case and my little knowledge about your project, it seems the third option is more suitable.


Cheers!

**Comment by alm0ra** [comment]
Hey @arash-d 

Thanks for suggesting solutions.

I'd be happy if you contribute by opening pull request if you have time. :) i prefer third way too.

**Comment by alm0ra** [comment]
fixed in[v0.1.56]( https://github.com/alm0ra/mockafka-py/releases/tag/v0.1.56)
